src/psyclone/psyir/transformations/nemo_sign_trans.py

```diff
@@ -0,0 +1,175 @@
+        res_sign = res_abs
+        tmp_sign = A
+        if (tmp_sign<0.0) res_sign=res_sign*-1.0
+        R= ... res_x ...
+
+        where A and B could be an arbitrarily complex expressions and
+        where ABS is replaced with inline code by the NemoAbsTrans
+        transformation.
+
+        A symbol table is required as the NEMO api does not currently
+        contain a symbol table and one is required in order to create
+        temporary variables whose names do not clash with existing
+        code. This non-standard argument is also the reason why this
+        transformation is currently limited to the NEMO api.
+
+        This transformation requires the operation node to be a
+        descendent of an assignment and will raise an exception if
+        this is not the case.
+
+        :param node: a SIGN BinaryOperation node.
+        :type node: :py:class:`psyclone.psyGen.BinaryOperation`
+        :param symbol_table: the symbol table.
+        :type symbol_table: :py:class:`psyclone.psyir.symbols.SymbolTable`
+        :param options: a dictionary with options for transformations.
+        :type options: dictionary of string:values or None
+
+        :returns: 2-tuple of new schedule and memento of transform.
+        :rtype: (:py:class:`psyclone.nemo.NemoInvokeSchedule`, \
+                 :py:class:`psyclone.undoredo.Memento`)
+
+        '''
+        self.validate(node, symbol_table)
+
+        schedule = node.root
+        memento = Memento(schedule, self, [node, symbol_table])
+
+        oper_parent = node.parent
+        assignment = node.ancestor(Assignment)
+        # Create two temporary variables.
+        res_var = symbol_table.new_symbol_name("res_sign")
+        symbol_table.add(DataSymbol(res_var, DataType.REAL))
+        tmp_var = symbol_table.new_symbol_name("tmp_sign")
+        symbol_table.add(DataSymbol(tmp_var, DataType.REAL))
+
+        # Replace operator with a temporary (res_var).
+        oper_parent.children[node.position] = Reference(res_var,
+                                                        parent=oper_parent)
+
+        # res_var=ABS(B)
+        lhs = Reference(res_var)
+        rhs = UnaryOperation.create(UnaryOperation.Operator.ABS,
+                                    node.children[1])
+        new_assignment = Assignment.create(lhs, rhs)
+        new_assignment.parent = assignment.parent
+        assignment.parent.children.insert(assignment.position, new_assignment)
+
+        # Replace the ABS intrinsic with inline code.
+        abs_trans = NemoAbsTrans()
+        abs_trans.apply(rhs, symbol_table)
+
+        # tmp_var=A
+        lhs = Reference(tmp_var)
+        new_assignment = Assignment.create(lhs, node.children[0])
+        new_assignment.parent = assignment.parent
+        assignment.parent.children.insert(assignment.position, new_assignment)
+
+        # if condition: tmp_var<0.0
+        lhs = Reference(tmp_var)
+        rhs = Literal("0.0", DataType.REAL)
```

Comment:
Please add a reference to the Issue I requested earlier to lift this assumption that the argument is REAL. Please also document this current limitation for the affected transformations.

Reply:
Yes done, although I hadn't documented it - but have now.